### PR TITLE
mlog: Add support for pager

### PR DESCRIPTION
**Pull request by xclaesse**
It is useful to redirect some outputs, such as "meson configure" to a pager (e.h. less). This is similar to most git commands.

### CODE REVIEW

**Comment by xclaesse** [comment]
Extended this PR to also print with colors.

image

**Comment by tristan957** [comment]
Can you add an option like `--no-pager` similar to Git?

**Comment by xclaesse** [comment]
> Can you add an option like `--no-pager` similar to Git?

done

**Comment by tristan957** [comment]
Only difference between Meson and Git now is that `--no-pager` sits on the root `git` parsing, while this sits on `meson configure`. Not sure which way is better.

**Comment by xclaesse** [comment]
I prefer it on the configure subcommand, most commands makes no sense to page, like configure.

File: mesonbuild/mconf.py [hunk 1]
```diff
@@ -302,14 +289,16 @@ def print_nondefault_buildtype_options(self):
...
+        mlog.log('               current   default')
         for m in mismatching:
-            print(f'{m[0]:21}{m[1]:10}{m[2]:10}')
+            mlog.log(f'{m[0]:21}{m[1]:10}{m[2]:10}')
 
 def run(options):
     coredata.parse_cmd_line_options(options)
     builddir = os.path.abspath(os.path.realpath(options.builddir))
+    if not options.no_pager:
+        mlog.start_pager()
```
Comment (anarazel) [inline mesonbuild/mconf.py:301]: Shouldn't a pager only be used if stdout is a tty?

Comment (xclaesse) [reply mesonbuild/mconf.py:301]: Good catch, and we already have the boilerplate for that, it's exactly what colorize_console() does.

Fixed.

**Comment by xclaesse** [comment]
I added a releases notes snippet.

**Comment by tristan957** [comment]
Looks good to me, but maybe @eli-schwartz has a second to give a little review.

**Comment by xclaesse** [comment]
We have a long backlog of PRs to review, we need to trust more reviewers, IMHO you're long enough contributor for this.

**Comment by tristan957** [comment]
Feel free to bring it up in the matrix channel

**Pull request merged**